File: Hard/2026-05-24-1340-Jump-Game-V/solution.java
import java.util.Arrays;

public class Solution {
    public int maxJumps(int[] arr, int d) {
        int n = arr.length;
        int[] dp = new int[n];
        Arrays.fill(dp, 1);
        
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j <= Math.min(i + d, n - 1); j++) {
                if (arr[j] < arr[i] && isValidJump(arr, i, j)) {
                    dp[j] = Math.max(dp[j], dp[i] + 1);
                }
            }
            for (int j = i - 1; j >= Math.max(i - d, 0); j--) {
                if (arr[j] < arr[i] && isValidJump(arr, i, j)) {
                    dp[j] = Math.max(dp[j], dp[i] + 1);
                }
            }
        }
        
        return Arrays.stream(dp).max().getAsInt();
    }
    
    private boolean isValidJump(int[] arr, int i, int j) {
        if (i < j) {
            for (int k = i + 1; k < j; k++) {
                if (arr[k] >= arr[i]) {
                    return false;
                }
            }
        } else {
            for (int k = j + 1; k < i; k++) {
                if (arr[k] >= arr[i]) {
                    return false;
                }
            }
        }
        return true;
    }
}

File: Hard/2026-05-24-1340-Jump-Game-V/solution.js
var maxJumps = function(arr, d) {
    const n = arr.length;
    const dp = new Array(n).fill(1);
    
    const canJump = (i, j) => {
        const [start, end] = i < j ? [i, j] : [j, i];
        for (let k = start + 1; k < end; k++) {
            if (arr[k] >= arr[i]) return false;
        }
        return true;
    };

    for (let i = 0; i < n; i++) {
        for (let j = 1; j <= d; j++) {
            if (i + j < n && arr[i] > arr[i + j] && canJump(i, i + j)) {
                dp[i + j] = Math.max(dp[i + j], dp[i] + 1);
            }
            if (i - j >= 0 && arr[i] > arr[i - j] && canJump(i, i - j)) {
                dp[i - j] = Math.max(dp[i - j], dp[i] + 1);
            }
        }
    }

    return Math.max(...dp);
};